### PR TITLE
Disable Debugger When Checkpointing Is Enabled

### PR DESCRIPTION
**Pull request by NihalHarish**
*Issue #, if available:*

*Description of changes:*
- We do not want debugger to run in the background if customers enable checkpointing.

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/estimator.py [hunk 1]
```diff
@@ -2147,6 +2147,11 @@ def __init__(
         self._hyperparameters = hyperparameters or {}
         self.checkpoint_s3_uri = checkpoint_s3_uri
         self.checkpoint_local_path = checkpoint_local_path
+
+        # Disable debugger if checkpointing is enabled by the customer
+        self.debugger_hook_config = \
```
Comment (ndodda-amazon) [inline src/sagemaker/estimator.py:2152]: If a user specifies a custom hook config while checkpointing is enabled, should we log or print a message saying that debugger is disabled?

File: tests/integ/test_debugger.py [hunk 2]
```diff
@@ -351,6 +351,50 @@ def test_mxnet_with_debugger_hook_config(
...
+        )
+
+        train_input = mx.sagemaker_session.upload_data(
+            path=os.path.join(data_path, "train"), key_prefix="integ-test-data/mxnet_mnist/train"
+        )
+        test_input = mx.sagemaker_session.upload_data(
+            path=os.path.join(data_path, "test"), key_prefix="integ-test-data/mxnet_mnist/test"
+        )
+
+        mx.fit({"train": train_input, "test": test_input})
```
Comment (ndodda-amazon) [inline tests/integ/test_debugger.py:392]: Do we actually need to train to test the behavior? Can't we simply inspect the training params after the estimator is created? Something like

```
assert "DebugHookConfig" not in mx.describe()`
```

**Review by ndodda-amazon — CHANGES_REQUESTED** [review]
Changes look good, left a few minor comments.

File: src/sagemaker/estimator.py (hunk 1 above)
Comment (NihalHarish) [reply src/sagemaker/estimator.py]: Done

File: tests/integ/test_debugger.py (hunk 2 above)
Comment (NihalHarish) [reply tests/integ/test_debugger.py]: Updated the test to not run an end-to-end training job

**Review by ndodda-amazon — DISMISSED** [review]
LGTM, thanks for doing this!

File: src/sagemaker/estimator.py [hunk 3]
```diff
@@ -2209,7 +2209,14 @@ def _validate_and_set_debugger_configs(self):
         ):
             self.debugger_hook_config = DebuggerHookConfig(s3_output_path=self.output_path)
         elif not self.debugger_hook_config:
-            self.debugger_hook_config = None
+            self.debugger_hook_config = False
```
Comment (ajaykarpur) [inline src/sagemaker/estimator.py:2212]: Why did this change from `None` to `False`?

Comment (NihalHarish) [reply src/sagemaker/estimator.py:2212]: The line above intends to disable the `self.debugger_hook_config`, setting it to `False` is a stronger indication than `None`

File: src/sagemaker/estimator.py [hunk 4]
```diff
@@ -2219,7 +2219,14 @@ def _validate_and_set_debugger_configs(self):
...
+
+        # Disable debugger if checkpointing is enabled by the customer
+        if self.checkpoint_s3_uri and self.checkpoint_local_path and self.debugger_hook_config:
+            logger.info(
+                "SM Debug Does Not Currently Support Training Jobs With Checkpointing Enabled"
```
Comment (ajaykarpur) [inline src/sagemaker/estimator.py:2227]: nit: Can you make this sentence case instead of capitalizing each word?

**Review by ajaykarpur — APPROVED** [review]

**Comment by NihalHarish** [comment]
Closing this since the duplicate was already merged with #2264

**Pull request closed without merging**